Nicely format Scribble Hub's author's notes & news

Epicpkmn11 opened the pull request:
This makes it so that instead of removing authors notes and leaving news boxes alone from Scribble Hub it instead formats them basically the same as AO3's chapter notes, with a bold title and then in a block quote. I also have it putting them in a `fff_chapter_notes` div, the same as AO3, so that custom CSS on that should work the same I think.

I tested with [The Walls of Anamoor](https://www.scribblehub.com/series/161834/the-walls-of-anamoor/) chapter 1 and [A Transformative Spark](https://www.scribblehub.com/series/161220/a-transformative-spark/) chapter 1 as they have two of news boxes and author's notes respectively.

<details><summary>Examples</summary>

![スクリーンショット 2020-10-02 10 47 50](https://user-images.githubusercontent.com/41608708/94943077-cddaa600-049c-11eb-8523-bfc54cc3c8f8.png)
![スクリーンショット 2020-10-02 10 48 03](https://user-images.githubusercontent.com/41608708/94943086-cfa46980-049c-11eb-9788-d12465cb1ea0.png)

</details>

---

Please let me know if you'd like me to change anything or so, I tried to stick to the code style of the file but I don't work with Python or this plugin stuff too much so I may have done something badly 😅

JimmXinu commented:
The AO3 adapter has configuration for excluding each of the different types of note:
```## AO3 chapters can include several different types of notes.  We've
## traditionally included them all in the chapter text, but this allows
## you to customize which you include.  Copy this parameter to your
## personal.ini and list the ones you don't want.
#exclude_notes:authorheadnotes,chaptersummary,chapterheadnotes,chapterfootnotes,authorfootnotes,inspiredlinks
```
I don't read scribblehub.com, so I don't care.  But if users complain, will you be willing to come back and make them configurable?

Epicpkmn11 commented:
> The AO3 adapter has configuration for excluding each of the different types of note:
> 
> ```
> ## traditionally included them all in the chapter text, but this allows
> ## you to customize which you include.  Copy this parameter to your
> ## personal.ini and list the ones you don't want.
> #exclude_notes:authorheadnotes,chaptersummary,chapterheadnotes,chapterfootnotes,authorfootnotes,inspiredlinks
> ```
> 
> I don't read scribblehub.com, so I don't care. But if users complain, will you be willing to come back and make them configurable?

Ah yeah, I can add that now if you'd like

Epicpkmn11 commented:
Alright, just added that, it seems to be working fine in calibre, I can't figure out how to test the CLI version 😅 but I think it should work there too.

I just kept the comment in the ini basically the same as the AO3 one, but I changed the names to be more fitting.

Epicpkmn11 commented:
Was doing some more thorough testing and apparently its possible for the news boxes to not have titles, as seen in [Twisted Destiny chapter 22](https://www.scribblehub.com/read/60207-twisted-destiny/chapter/107917/), so I've added checks for the news blocks and author's notes titles and bodies, I'm not sure if its possible to leave the body out as this chapter has some weird triple stack of news boxes going on to be missing titles, but as its possible to leave the title out I figured I'd play it safe.

JimmXinu commented on fanficfare/adapters/adapter_scribblehubcom.py:
```diff
@@ -270,8 +270,64 @@ def getChapterText(self, url):
+        if 'authornotes' in exclude_notes:
+            # Remove author's notes
+            # There can be multiple in one chapter, so loop until none are left
+            while div.find('div', {'class' : 'wi_authornotes'}):
```
This would more efficient as:
```
for antag in div.find_all('div', {'class' : 'wi_authornotes'}):
    antag.decompose()
```
As it is, you're searching the soup tree over and over again.

If you are more comfortable with CSS selectors, BeautifulSoup also has `select()` and `select_one()` methods:
```
for antag in div.select('div.wi_authornotes'):
    antag.decompose()
```
(Many of the FFF supported sites predate those methods in BS and I was more comfortable with `find()` and `find_all()` for a long time, which is why you see more `find()`s in FFF.)

Epicpkmn11 replied:
Ah yeah, that's much cleaner, thanks for pointing it out

JimmXinu commented on fanficfare/adapters/adapter_scribblehubcom.py:
```diff
@@ -270,8 +270,64 @@ def getChapterText(self, url):
+            # Reformat the author's notes
+            # There can be multiple in one chapter, so loop until none are left
+            while div.find('div', {'class' : 'wi_authornotes'}):
+                author_notes = div.find('div', {'class' : 'wi_authornotes'})
```
Here too:
`for author_notes in div.find_all('div', {'class' : 'wi_authornotes'}):`

Also, if somebody later stopped clearing `'wi_authornotes'` from `author_notes['class']` you'd end up with an infinite loop.  Best to avoid the possibility.

Ditto for the news searches below, too.

JimmXinu reviewed (changes requested):
Please change the loop structures as comments.  Thanks!
